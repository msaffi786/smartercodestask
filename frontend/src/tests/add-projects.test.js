import React from "react";
import {
  render,
  screen,
  cleanup,
  fireEvent,
  waitFor,
} from "@testing-library/react";
import renderer from "react-test-renderer";
import { BrowserRouter } from "react-router-dom";
import axios from "axios";
import AddProject from "../components/add-project.component";
axios.defaults.adapter = require("axios/lib/adapters/http");
afterEach(cleanup);
// axios.defaults.adapter = require('axios/lib/adapters/http');
test("renders Welcome", () => {
  const { asFragment } = render(
    <BrowserRouter>
      <AddProject />
    </BrowserRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});


test("add new project", async () => {
  const { getByText, getByLabelText, rerender, getByRole } = render(
    <BrowserRouter>
      <AddProject />
    </BrowserRouter>
  );

  getByLabelText("Title");
  getByLabelText("Description");
  const inputName = getByLabelText("Title");
  fireEvent.change(inputName, { target: { value: "new jest title" } });
  const inputAge = getByLabelText("Description");
  fireEvent.change(inputAge, { target: { value: "new jest description" } });
  const button = screen.getByText("Submit");
  fireEvent.click(button);
  //   await waitFor(() => fireEvent.click(button)
  expect(inputName.innerText).toBe(undefined);
});
