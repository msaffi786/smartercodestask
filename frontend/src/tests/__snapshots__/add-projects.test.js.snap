// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Welcome 1`] = `
<DocumentFragment>
  <div
    class="card"
  >
    <div
      class="card-body"
    >
      <div
        class="container"
      >
        <div
          class="row"
        >
          <div
            class="col-4"
          >
            <div
              class="card"
            >
              <div
                class="card-body"
              >
                <div>
                  <div
                    class="navbar-nav ml-auto"
                  >
                    <li
                      class="nav-item"
                    >
                      <a
                        class="nav-link"
                        href="/login"
                      >
                        Login
                      </a>
                    </li>
                    <li
                      class="nav-item"
                    >
                      <a
                        class="nav-link"
                        href="/register"
                      >
                        Sign Up
                      </a>
                    </li>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="col-8"
          >
            <div
              class="container"
            >
              <div
                class="submit-form"
              >
                <div>
                  <div
                    class="form-group"
                  >
                    <label
                      for="title"
                    >
                      Title
                    </label>
                    <input
                      class="form-control"
                      id="title"
                      name="title"
                      required=""
                      type="text"
                      value=""
                    />
                  </div>
                  <div
                    class="form-group"
                  >
                    <label
                      for="description"
                    >
                      Description
                    </label>
                    <input
                      class="form-control"
                      id="description"
                      name="description"
                      required=""
                      type="text"
                      value=""
                    />
                  </div>
                  <div
                    class="App"
                  >
                    <h3>
                      Select Toppings
                    </h3>
                    <ul
                      class="toppings-list"
                    >
                      <li>
                        <div
                          class="toppings-list-item"
                        >
                          <div
                            class="left-section"
                          >
                            <input
                              id="custom-checkbox-0"
                              name="Capsicum"
                              type="checkbox"
                              value="Capsicum"
                            />
                            <label
                              for="custom-checkbox-0"
                            >
                              Capsicum
                            </label>
                          </div>
                        </div>
                      </li>
                      <li>
                        <div
                          class="toppings-list-item"
                        >
                          <div
                            class="left-section"
                          >
                            <input
                              id="custom-checkbox-1"
                              name="Paneer"
                              type="checkbox"
                              value="Paneer"
                            />
                            <label
                              for="custom-checkbox-1"
                            >
                              Paneer
                            </label>
                          </div>
                        </div>
                      </li>
                      <li>
                        <div
                          class="toppings-list-item"
                        >
                          <div
                            class="left-section"
                          >
                            <input
                              id="custom-checkbox-2"
                              name="Red Paprika"
                              type="checkbox"
                              value="Red Paprika"
                            />
                            <label
                              for="custom-checkbox-2"
                            >
                              Red Paprika
                            </label>
                          </div>
                        </div>
                      </li>
                      <li>
                        <div
                          class="toppings-list-item"
                        >
                          <div
                            class="left-section"
                          >
                            <input
                              id="custom-checkbox-3"
                              name="Onions"
                              type="checkbox"
                              value="Onions"
                            />
                            <label
                              for="custom-checkbox-3"
                            >
                              Onions
                            </label>
                          </div>
                        </div>
                      </li>
                      <li>
                        <div
                          class="toppings-list-item"
                        >
                          <div
                            class="left-section"
                          >
                            <input
                              id="custom-checkbox-4"
                              name="Extra Cheese"
                              type="checkbox"
                              value="Extra Cheese"
                            />
                            <label
                              for="custom-checkbox-4"
                            >
                              Extra Cheese
                            </label>
                          </div>
                        </div>
                      </li>
                      <li>
                        <div
                          class="toppings-list-item"
                        >
                          <div
                            class="left-section"
                          >
                            <input
                              id="custom-checkbox-5"
                              name="Baby Corns"
                              type="checkbox"
                              value="Baby Corns"
                            />
                            <label
                              for="custom-checkbox-5"
                            >
                              Baby Corns
                            </label>
                          </div>
                        </div>
                      </li>
                      <li>
                        <div
                          class="toppings-list-item"
                        >
                          <div
                            class="left-section"
                          >
                            <input
                              id="custom-checkbox-6"
                              name="Mushroom"
                              type="checkbox"
                              value="Mushroom"
                            />
                            <label
                              for="custom-checkbox-6"
                            >
                              Mushroom
                            </label>
                          </div>
                        </div>
                      </li>
                    </ul>
                  </div>
                  <button
                    class="btn btn-success"
                  >
                    Submit
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
